runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/emote/STASHUnit.java

```diff
@@ -31,123 +31,127 @@
+	BY_THE_BEAR_CAGE_IN_VARROCK_PALACE_GARDENS(NullObjectID.NULL_29055, STASHBuildHelper.MASTER.getConstructionLevelRequired(), STASHBuildHelper.MASTER.getMaterialsRequired(), new WorldPoint(3232, 3494, 0)),
+	OUTSIDE_THE_WILDERNESS_AXE_HUT(NullObjectID.NULL_29056, STASHBuildHelper.MASTER.getConstructionLevelRequired(), STASHBuildHelper.MASTER.getMaterialsRequired(), new WorldPoint(3186, 3958, 0)),
+	TOP_FLOOR_OF_THE_YANILLE_WATCHTOWER(NullObjectID.NULL_29057, STASHBuildHelper.MASTER.getConstructionLevelRequired(), STASHBuildHelper.MASTER.getMaterialsRequired(), new WorldPoint(2930, 4718, 2)),
+	DEATH_ALTAR(NullObjectID.NULL_29058, STASHBuildHelper.MASTER.getConstructionLevelRequired(), STASHBuildHelper.MASTER.getMaterialsRequired(), new WorldPoint(2210, 4842, 0)),
+	BEHIND_MISS_SCHISM_IN_DRAYNOR_VILLAGE(NullObjectID.NULL_29059, STASHBuildHelper.MASTER.getConstructionLevelRequired(), STASHBuildHelper.MASTER.getMaterialsRequired(), new WorldPoint(3095, 3254, 0)),
+	NORTHWESTERN_CORNER_OF_THE_ENCHANTED_VALLEY(NullObjectID.NULL_29060, STASHBuildHelper.MASTER.getConstructionLevelRequired(), STASHBuildHelper.MASTER.getMaterialsRequired(), new WorldPoint(3022, 4517, 0)),
+	NORTH_OF_MOUNT_KARUULM(NullObjectID.NULL_34647, STASHBuildHelper.MEDIUM.getConstructionLevelRequired(), STASHBuildHelper.MEDIUM.getMaterialsRequired(), new WorldPoint(1308, 3840, 0)),
+	GYPSY_TENT_ENTRANCE(NullObjectID.NULL_34736, STASHBuildHelper.BEGINNER.getConstructionLevelRequired(), STASHBuildHelper.BEGINNER.getMaterialsRequired(), new WorldPoint(3206, 3422, 0)),
+	FINE_CLOTHES_ENTRANCE(NullObjectID.NULL_34737, STASHBuildHelper.BEGINNER.getConstructionLevelRequired(), STASHBuildHelper.BEGINNER.getMaterialsRequired(), new WorldPoint(3209, 3416, 0)),
+	BOB_AXES_ENTRANCE(NullObjectID.NULL_34738, STASHBuildHelper.BEGINNER.getConstructionLevelRequired(), STASHBuildHelper.BEGINNER.getMaterialsRequired(), new WorldPoint(3233, 3200, 0)),
+	CRYSTALLINE_MAPLE_TREES(NullObjectID.NULL_34953, STASHBuildHelper.MASTER.getConstructionLevelRequired(), STASHBuildHelper.MASTER.getMaterialsRequired(), new WorldPoint(2213, 3427, 0)),
+	CHARCOAL_BURNERS(NullObjectID.NULL_41758, STASHBuildHelper.ELITE.getConstructionLevelRequired(), STASHBuildHelper.ELITE.getMaterialsRequired(), new WorldPoint(1712, 3470, 0)),
 	;
 
 	private final int objectId;
 	private final WorldPoint[] worldPoints;
+	private final String stashLevelBuildRequirements;
+	private final String stashItemBuildRequirements;
 
-	STASHUnit(int objectId, WorldPoint... worldPoints)
+	STASHUnit(int objectId, String stashLevelBuildRequirements, String stashBuildRequirements, WorldPoint... worldPoints)
 	{
 		this.objectId = objectId;
+		this.stashLevelBuildRequirements = stashLevelBuildRequirements;
+		this.stashItemBuildRequirements = stashBuildRequirements;
```

Comment:
[Lines 147-148](https://github.com/runelite/runelite/pull/15574/files#diff-167d262ac5e97af29d9a58235f5251c3e99bcc2953004f244f88cdd026d80988R147-R148) - Why aren't you using your `STASHBuildHelper` enum here?

You can simplify the additional constructor args passed to only be an enum and then apply your getter logic to the enum itself rather than bloating the constructors with it - for example, [Line 142](https://github.com/runelite/runelite/pull/15574/files#diff-167d262ac5e97af29d9a58235f5251c3e99bcc2953004f244f88cdd026d80988R142) can go from
```CHARCOAL_BURNERS(NullObjectID.NULL_41758, STASHBuildHelper.ELITE.getConstructionLevelRequired(), STASHBuildHelper.ELITE.getMaterialsRequired(), new WorldPoint(1712, 3470, 0)),```
to
```CHARCOAL_BURNERS(NullObjectID.NULL_41758, STASHBuildHelper.ELITE, new WorldPoint(1712, 3470, 0))```

Reply:
Thank you for pointing this out. I will make these changes to my code to clean it up.